modules/auxiliary/scanner/smb/smb_ms17_010.rb

```diff
@@ -154,6 +157,8 @@ def run_host(ip)
     ensure
       disconnect
     end
+
+    vulnerable
```

Comment:
Not sure how clear or portable this is since in other languages variables declared/first used inside a lexical scope like our `if` expression are not usable from outside.

Reply:
This is the weakest link, and I'm currently reworking it.